build_tools/ci/cpu_comparison/run.py

```diff
@@ -129,17 +176,17 @@ def get_filename(self, config):
 
 
 class ConvolutionFromTemplate(BaseTest):
-    def __init__(self, params):
-        super().__init__()
-        self.generator = ConvolutionMlirGenerator(**params)
-        params = self.generator.params
-        conv_type = params["conv_type"]
-        N = params["N"]
-        IW = params["IW"]
-        in_type = params["input_element_type"]
-        out_type = params["output_element_type"]
+    def __init__(
+        self,
+        generator,
+        test_params=None,
+    ):
+        super().__init__(
+            name=f"{generator.params['conv_type']}_{generator.params['N']}_{generator.params['IW']}_{generator.params['input_element_type']}_{generator.params['output_element_type']}",
+            test_params=test_params if test_params is not None else TestParams(),
```

Comment:
No need for this, you can just forward `test_params`. Same for all the other subclasses down below.
```suggestion
            test_params=test_params,
```